src/transport.rs

```diff
@@ -11,51 +11,689 @@
+            send_to_inner_with_status(
+                &self.tx_to_event_loop,
+                TransportCommand::RegisterListener(
+                    src.clone(),
+                    Some(sink.clone()),
+                    message_type,
+                    msg_handler,
+                    tx,
+                ),
+            )
+            .await?;
+            await_internal_function(UP_CLIENT_VSOMEIP_FN_TAG_REGISTER_LISTENER_INTERNAL, rx)
+                .await?;
+        }
+        Ok(())
+    }
+
+    async fn register_listener_id_with_listener(
+        listener_id: usize,
+        listener: Arc<dyn UListener>,
+    ) -> Result<(), UStatus> {
+        LISTENER_REGISTRY
+            .write()
+            .await
+            .insert(listener_id, listener.clone())
+            .map(|_| {
+                Err(UStatus::fail_with_code(
+                    UCode::INTERNAL,
+                    "Unable to register the same listener_id and listener twice",
+                ))
+            })
+            .unwrap_or(Ok(()))?;
+        Ok(())
+    }
+
+    async fn register_point_to_point_listener(
+        &self,
+        listener: &Arc<dyn UListener>,
+    ) -> Result<(), UStatus> {
+        let Some(config_path) = &self.config_path else {
+            let err_msg = "No path to a vsomeip config file was provided";
+            error!("{err_msg}");
+            return Err(UStatus::fail_with_code(UCode::NOT_FOUND, err_msg));
+        };
+
+        let application_configs = extract_applications(config_path)?;
+        trace!("Got vsomeip application_configs: {application_configs:?}");
+
+        {
+            let mut point_to_point_listener = self.point_to_point_listener.write().await;
+            if point_to_point_listener.is_some() {
+                return Err(UStatus::fail_with_code(
+                    UCode::ALREADY_EXISTS,
+                    "We already have a point-to-point UListener registered",
+                ));
+            }
+            *point_to_point_listener = Some(listener.clone());
+            trace!("We found a point-to-point listener and set it");
+        }
+
+        for app_config in &application_configs {
+            let (tx, rx) = oneshot::channel();
+            send_to_inner_with_status(
+                &self.tx_to_event_loop,
+                TransportCommand::InitializeNewApp(app_config.id, app_config.name.clone(), tx),
+            )
+            .await?;
+            await_internal_function(
+                &format!("Initializing point-to-point listener apps. ApplicationConfig: {:?} app_config.id: {} app_config.name: {}",
+                app_config, app_config.id, app_config.name),
+                rx,
+            )
+            .await?;
+
+            let listener_id = find_available_listener_id().await?;
+            let comp_listener = ComparableListener::new(listener.clone());
+
+            let src = any_uuri();
+            // TODO: How to explicitly handle instance_id?
```

Comment:
Given that we cannot set the instance_id within the vsomeip config file, how do we handle this? For now I'm setting the upper four bytes to zero.

Tagging @stevenhartley / @int0x27